static/app/views/automations/components/actionNodeList.tsx

```diff
@@ -24,14 +30,51 @@ export default function ActionNodeList({
   group,
   placeholder,
   actions,
-  availableActions,
   onAddRow,
   onDeleteRow,
   updateAction,
 }: ActionNodeListProps) {
-  const options = Array.from(actionNodesMap)
-    .filter(([value]) => availableActions.some(action => action.type === value))
-    .map(([value, {label}]) => ({value, label}));
+  const {data: availableActions = []} = useAvailableActionsQuery();
+  const actionHandlerMapRef = useRef(new Map());
+
+  const options = useMemo(() => {
+    const typeOptionsMap = new Map<
+      ActionGroup,
+      Array<{label: string; value: ActionHandler}>
+    >();
```

Comment:
doesn't really make sense to use a map here since you're setting three keys. would just initialize them?